### PR TITLE
Global hotkey functionality for Windows

### LINKED ISSUE #24: Implement ability to cancel taking a cropped screenshot
For example using the <kbd>Esc</kbd> key

### PR DESCRIPTION
**Pull request by denniseffing**
This pull request includes hotkey functionality for Windows. It also introduces the ability to cancel the screen capture with windows_tk and fixes a bug when cancelling screen capture.

~~Soon&trade;: Parsing of composite hotkeys.~~ Done.

References #17 
Closes #24

### CODE REVIEW

**Comment by dtext** [comment]
You did not include the default settings for the hotkeys in `instantshare.default` yet.
Also, please merge the latest changes back into your feature branch before you do that.

**Comment by dtext** [comment]
The `tools` package is intended for more generic tools that can easily be reused. To avoid coupling with other functionality, I would suggest an interface like

```
bind_hotkey(hotkey: str, f: function)
unbind_hotkey(hotkey: str)
```

along with an HotkeyInUseError or something like that.

**Comment by dtext** [comment]
I just thought about this: What if the user does not want to use hotkeys at all? We need to consider that as well. Maybe even make that the default?

**Comment by denniseffing** [comment]
Well, we can certainly implement a config option to disable hotkeys but in my opinion it should be on by default.

**Comment by dtext** [comment]
I was thinking of leaving the config option empty. That would give users the option to disable the hotkey for certain things, while still having hotkeys for other things.

The default does not really matter in the end, I was just thinking it could be good to leave them off by default so they would not conflict with other Hotkeys people have setup. But I guess supplying a default that people can change as needed has its advantages as well.

Which makes me think of another thing: What happens when Hotkeys are already in use by another application? And what happens if the user defines the same hotkey for multiple functions? I think we may be missing some error handling here.

**Comment by denniseffing** [comment]
As of now, you can just omit individual hotkey options in the config file if you don't need them. If you want to disable the hotkeys completely, just omit every hotkey value.

When hotkeys are already in use by another application, it triggers the hotkey action of instantshare as well as the hotkey action of the application. With the current implementation, there is no way to determine if another application also listens on the configured hotkey.

instantshare warns the user (via log) if he defines the same hotkey for multiple functions. It also warns him if he specified an invalid hotkey.

File: src/cli/gui.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+from tools.config import CONFIG
+
+
+def main(argv):
+    args = docopt(__doc__, argv)
+    from gui.traymenu import Tray
+    from tools.hotkey import Hotkey, HotkeyInUseError, InvalidHotkeyError
+    from tools.toolbox import delay_execution
+    from queue import Queue
```
Comment (dtext) [inline src/cli/gui.py]: Is there a reason to import this in the main function instead of the top of the module?
Would probably move this to Line 16.

**Comment by dtext** [comment]
Added one small line comment, but everything else looks good now! :+1:

**Pull request merged**